src/constraint.rs

```diff
@@ -703,3 +715,57 @@ impl ConstraintValidator for FieldsConstraint {
         Ok(())
     }
 }
+
+/// Implements Ion Schema's `contains` constraint
+/// [contains]: https://amzn.github.io/ion-schema/docs/spec.html#contains
+#[derive(Debug, Clone, PartialEq)]
+pub struct ContainsConstraint {
+    // TODO: convert this into a HashSet once we have an implementation of Hash for OwnedElement in ion-rust
+    // Reference ion-rust issue: https://github.com/amzn/ion-rust/issues/220
+    values: Vec<OwnedElement>,
+}
+
+impl ContainsConstraint {
+    pub fn new(values: Vec<OwnedElement>) -> Self {
+        Self { values }
+    }
+}
+
+impl ConstraintValidator for ContainsConstraint {
+    fn validate(&self, value: &OwnedElement, type_store: &TypeStore) -> ValidationResult {
+        // get the expected values from the contains constraint
+        let mut expected_values = self.values.iter();
+
+        // Check for null sequence
+        if value.is_null() {
+            return Err(Violation::new(
+                "contains",
+                ViolationCode::TypeMismatched,
+                &format!("Null sequence not allowed for contains constraint"),
```

Comment:
minor suggestion—The wording here initially had me confused whether this was validating the value or validating that the schema was correct. I think it might be a little more clear to make the phrasing here be similar to the phrasing of the type mismatch on Line 754.
```suggestion
                &format!("expected a sequence found {}", value),
```